src/include/miopen/softmax/problem_description.hpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <miopen/problem_description_base.hpp>
+#include <miopen/tensor.hpp>
+
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace softmax {
+
+int nextPow2(int v);
+
+void getParams(const TensorDescriptor& in_yDesc, 
+                miopenSoftmaxMode_t in_mode, 
+                int& out_n, 
+                int& out_c, 
+                int& out_h, 
+                int& out_w, 
+                int& out_grid_size, 
+                int& out_spatial_dim,
+                int& out_vector_size,
+                int& out_num_batch,
+                bool& out_usefp16,
+                bool& out_usefp32,
+                std::vector<size_t>& out_vld,
+                std::vector<size_t>& out_vgd,
+                size_t& out_workgroups,
+                int& out_batch_size,
+                int& out_u_batch_size);
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription( const void* alpha_,
+                        const void* beta_,
+                        const TensorDescriptor& xDesc_,
+                        const TensorDescriptor& yDesc_,
+                        miopenSoftmaxAlgorithm_t algorithm_,
+                        miopenSoftmaxMode_t mode_) :
+        alpha(alpha_), beta(beta_), xDesc(xDesc_), yDesc(yDesc_), algorithm(algorithm_), mode(mode_)
+    {
+    }
```

Comment:
These checks  are present in  miopenStatus_t SoftmaxForward(...) function
But I will add additional alpha and beta checks.

Reply:
@Vsevolod1983 It seems better to check and block construction of invalid ProblemDescriptions rather than implement the same checks here and there in the code. So please,

> These checks are present in miopenStatus_t SoftmaxForward(...) function

Move these into ctor of PD. Test manually if it works properly.

Reply:
@atamazov 
We cannot do it in all cases, because several parameters are not part of ProblemDescription. For example x and y - are data params, they are not included in ProblemDescription ctor, but we need to check them as well (and throws an exception if they are invalid).
Also, we have a case when we throw an exception only after we calculate a lot of intermediate params. So, its not a part of PD ctor as well.

Reply:
@Vsevolod1983 
> Also, we have a case when we throw an exception only after we calculate a lot of intermediate params. So, its not a part of PD ctor as well.

Does this case indicate invalid problem?

Reply:
@atamazov 
Hi, I moved some checks to ctor of PD class. One check stayed on solver side though. Please have a look.

Also I changed network config, will appreciate if you have a look as well.